airflow/providers/microsoft/azure/hooks/data_factory.py

```diff
@@ -828,7 +833,14 @@ def wait_for_pipeline_run_status(
             # Wait to check the status of the pipeline run based on the ``check_interval`` configured.
             time.sleep(check_interval)
 
-            pipeline_run_status = self.get_pipeline_run_status(**pipeline_run_info)
+            try:
+                pipeline_run_status = self.get_pipeline_run_status(**pipeline_run_info)
+                executed_after_token_refresh = True
+            except Exception:
```

Comment:
Can we try using a specific exception instead of using `Exception` class which is too generic

Reply:
Sure, I used `Exception` because of the original design of `get_pipeline_run` and `get_adf_pipeline_run_status` methods. They caught all exceptions and raised them as `AirflowException`, which makes it a bit complicated to handle the exception in the triggerer. Thus, I unwrapped the try-catch block for these 2 methods. I also traced the code to confirm these methods are not used elsewhere. Thus, it seems to me that these try-catch blocks might not be necessary and could make exception handling complicate